### PR TITLE
Replacing `useMediaQuery` with custom hook (fixes #2177)

### LINKED ISSUE #2177: Frontend renders componentes too early
### Description of the bug

After upgrading the frontend to v4 we have noticed that some components seem to be renderering before everything is ready. The Likert component is rendered (when in mobile / narrow width view) as the fullscreen variant for a few milliseconds before it   renders as normal radio groups. We also noticed that the progress circle renders the state 0/n pages when navigating from a stateless part to the statefull part, this didn't happen with v3.

### Steps To Reproduce

1. Go to https://udir.apps.tt02.altinn.no/udir/pbu-dev-demo?survey=survey-6-aarstrinn5
2. Make the viewport narrow (view as iPad Air for instance), so that any likert component is rendered as a radio group instead of a table
3. Navigate to page 3/18, here we will see the short flicker with the wide version of the Likert component

### Additional Information

The video illustrates the Likert component issue.
https://github.com/Altinn/app-frontend-react/assets/3714805/1d31a672-8938-4482-a90c-41d2bac85fd8

### PR DESCRIPTION
**Pull request by olemartinorg**
## Description

Turns out the MUI `useMediaQuery()` hook might be a big cannon for our use-cases, and it failed to get the correct initial state (which causes the UI to render in desktop mode once before rendering in mobile mode). I replaced it with a custom hook. I haven't added any tests for this, as it's typically difficult to _test that something doesn't happen_, and I'm not that afraid this will regress any time soon.

## Related Issue(s)

- closes #2177

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [x] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [x] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

File: src/hooks/useIsMobile.ts
Comment (cammiida) [inline src/hooks/useIsMobile.ts:1]: Ikke noe du har skrevet nå, men kan det være en ide å endre navnet på fila? Siden den inkluderer hooks for alle "devices"? 😊

File: src/hooks/useIsMobile.ts [hunk 1]
```diff
@@ -1,27 +1,50 @@
...
-  return useMediaQuery(`(max-width:${breakpoints.tablet}px)`);
+  return useBrowserWidth(conditionIsMobileOrTablet);
+}
+
+function useBrowserWidth(condition: Condition) {
+  const [state, setState] = useState(condition(window.innerWidth));
+
+  const handleResize = useCallback(() => {
+    setState(condition(window.innerWidth));
+  }, [condition]);
```
Comment (cammiida) [inline src/hooks/useIsMobile.ts:42]: Du kan putte denne inni `useEffect`-en, så slipper du også `useCallback`-en 👌

**Review by cammiida — APPROVED** [review]
💯

File: src/hooks/useIsMobile.ts (hunk 1 above)
Comment (olemartinorg) [reply src/hooks/useIsMobile.ts:42]: Nice, enig! :raised_hands:

File: src/hooks/useIsMobile.ts
Comment (olemartinorg) [reply src/hooks/useIsMobile.ts:1]: Ja, enig - jeg endrer den til `useDeviceWidths.ts` heller. :+1:

**Pull request merged**